src/raft.rs

```diff
@@ -851,9 +859,15 @@ impl<T: Storage> Raft<T> {
 
     /// Sends an append RPC with new entries (if any) and the current commit index to the given
     /// peer.
-    pub fn send_append(&mut self, to: u64) {
+    pub fn send_append(&mut self, to: u64, exhaust: bool) {
```

Comment:
How about using a new method? So you don't need to change so many codes and the new method can be marked as private.